File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::vector<int> result;
        std::unordered_map<int, int> mp; 
        std::priority_queue<std::pair<int, int>> q;

        for (int num : nums) { mp[num]++; }
        for (const auto &pair : mp) { q.push({pair.second, pair.first}); }
        while (result.size() < k) {
            result.emplace_back(q.top().second);
            q.pop();
        }

        return result;
    }
};

File: Data Structures & Algorithms/crawler-log-folder/submission-1.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        st = []

        for log in logs:
            if log == "./":
                continue
            elif log == "../":
                if st:
                    st.pop()
            else:
                st.append(log)

        return len(st)